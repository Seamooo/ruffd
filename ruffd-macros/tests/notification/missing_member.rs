use ruffd_macros::notification;

#[notification(open_buffers, bad_struct_member)]
async fn some_notification(_params: i32) -> Result<(), ruffd_types::RpcError> {
    let _open_buffers = open_buffers;
    Ok(())
} 


fn main() {}

