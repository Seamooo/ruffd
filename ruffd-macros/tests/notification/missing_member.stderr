error[E0609]: no field `bad_struct_member` on type `ServerStateHandles<'_>`
 --> tests/notification/missing_member.rs:3:30
  |
3 | #[notification(open_buffers, bad_struct_member)]
  |                              ^^^^^^^^^^^^^^^^^ unknown field
  |
  = note: available fields are: `project_root`, `open_buffers`, `capabilities`, `settings`

error[E0609]: no field `bad_struct_member` on type `ServerStateLocks`
 --> tests/notification/missing_member.rs:3:30
  |
3 | #[notification(open_buffers, bad_struct_member)]
  |                              ^^^^^^^^^^^^^^^^^ unknown field
  |
  = note: available fields are: `project_root`, `open_buffers`, `capabilities`, `settings`

error[E0609]: no field `bad_struct_member` on type `ruffd_types::tokio::sync::MutexGuard<'_, ServerState>`
 --> tests/notification/missing_member.rs:3:30
  |
3 | #[notification(open_buffers, bad_struct_member)]
  |                              ^^^^^^^^^^^^^^^^^ unknown field
